client/src/app/site/motions/components/motion-import-list/motion-import-list.component.ts

```diff
@@ -0,0 +1,265 @@
+        super(titleService, translate, matSnackBar);
+        this.initTable();
+        this.selectedEncoding = this.encodings[0];
+        this.importer.errorEvent.subscribe( message => this.raiseError(message) );
+    }
+
+    /**
+     * Children need to call this in their init-function.
+     * Calling these three functions in the constructor of this class
+     * would be too early, resulting in non-paginated tables
+     */
+    public initTable(): void {
+        this.dataSource = new MatTableDataSource();
+        const me = this;
+        this.setFilter();
+        this.importer.getNewEntries().subscribe(newEntries => {
+            me.dataSource.data = newEntries;
+            if (newEntries.length){
+                me.hasFile = true;
+            }
+        });
+    }
+
+    /**
+     * Returns the table column definition
+     */
+    public getColumnDefinition(): string[]{
+        return ['status'].concat(this.importer.expectedHeader);
+    }
+
+    /**
+     * toggle for 'done' checkbox
+     * TODO: more angular-material way to do this
+     */
+    public toggleHiddenDone() : void {
+        this.hidden.done = !this.hidden.done;
+        this.setFilter();
+    }
+
+    /**
+     * toggle for 'errors' checkbox
+     * TODO: more angular-material way to do this
+     */
+    public toggleHiddenErrors() : void {
+        this.hidden.errors = !this.hidden.errors;
+        this.setFilter();
+    }
+
+    /**
+     * triggers the importer's onSelectFile
+     */
+    public onSelectFile(event: any): void {
+        this.importer.onSelectFile(event, this.selectedEncoding);
+    }
+
+    /**
+     * Triggers the importer's import
+     */
+    public async doImport(): Promise<void> {
+        await this.importer.doImport();
+        this.setFilter();
+    }
+
+    /**
+     * Updates and manually triggers the filter function.
+     * See {@link hidden} for options
+     * (changed from default mat-table filter)
+     */
+    private setFilter(): void {
+        const me = this;
+        this.dataSource.filter = '';
+        this.dataSource.filterPredicate = function (
+            data: NewMotionEntry, filter: string ): boolean
+        {
+            if (me.hidden.done && data.status_done) {
+                return false;
+            }
+            if (me.hidden.errors) {
+                if (!data.newViewMotion.isValidMotion || data.updateModel.length) {
+                    return false;
+                }
+            }
+            return true;
+        };
+        this.dataSource.filter = 'X'; // TODO: This is just a bogus non-null string to trigger the filter
+    }
+
+    /**
+     * Returns a reason on why an import cannot be done automatically
+     * (empty if there is no reason)
+     * @param row
+     */
+    public getInvalidReason(row: NewMotionEntry): string {
+        if (!row.newMotion.title) {
+            return this.invalidReasons.title;
+        }
+        if (!row.newMotion.text) {
```

Comment:
Make make `else if` out of this and if `this.invalidReasons` is not used anywhere else, move the strings into this function and remove `this.invalidReasons`.